arangod/Aql/Optimizer/Rule/OptimizerRulesCluster.cpp

```diff
@@ -495,7 +495,8 @@ bool substituteClusterMultipleDocumentInsertOperations(
   }
 
   auto* enumerateNode = ExecutionNode::castTo<EnumerateListNode const*>(dep);
-  if (enumerateNode->outVariable() != mod->inVariable()) {
+  TRI_ASSERT(enumerateNode->getMode() == EnumerateListNode::kEnumerateArray);
```

Comment:
This assertion is puzzling. I can't prove that it is wrong, however, I also can't prove that it is correct. It is more accidentally correct. Instead of asserting, I'd prefer to check, and if the mode is object, then bail out and not optimize.